kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeResultGenerator.kt

```diff
@@ -245,18 +247,31 @@ data class TypeResultGenerator(val options: Options, val implicitAccessor: Acces
     private val TypeName.isTypeAlias: Boolean
         get() = tag(TypeAliasTag::class) != null
 
-    private fun TypeResult.Container.generate(): CodeBlock {
+    private fun TypeResult.SetContainer.generate(): CodeBlock {
         return CodeBlock.builder().apply {
-            add("$creator(")
-            add("\n⇥")
-            args.forEachIndexed { index, arg ->
-                if (index != 0) {
-                    add(",\n")
+            beginControlFlow("buildSet(%L)", args.size)
+            withIndent {
+                for (arg in args) {
+                    if (arg.key.type.isSet()) {
```

Comment:
actually I think i'll need a different check here. `isSet` wouldn't work for a `Set<Set<T>>` binding.